### ISSUE TITLE
Version 1.12.x Introduces a Dependency on a Vulnerable Version of Log4j.

### ISSUE THREAD

**Issue by scottweaver** [scalameta/metals-vscode#900]
**Describe the bug**

The newest version of metals-vscode 1.12.0, introduces a dependency on a vulnerable version of log4j, 2.8.1.

This breaks the Metals extension for any user that is sitting behind a repository proxy e.g. Artifactory, that actively disallows downloading known vulnerable versions of Log4j 

**To Reproduce** Steps to reproduce the behavior:

1. Install VS Code 1.12.x
2. Attempt import build.
3. Encounter the following failure:
```
2022.02.24 15:53:55 ERROR > No server running at 127.0.0.1:8212, let's fire one...
2022.02.24 15:53:55 ERROR > Resolving ch.epfl.scala:bloop-frontend_2.12:1.4.12...
2022.02.24 15:53:55 ERROR > error: Unexpected error when resolving Bloop server via coursier!
2022.02.24 15:53:55 ERROR > error: Error downloading org.apache.logging.log4j:log4j-core:2.8.1
2022.02.24 15:53:55 ERROR >   not found: https://artifactory.mydomain.com/artifactory/maven-internalfacing/org/apache/logging/log4j/log4j-core/2.8.1/log4j-core-2.8.1.pom
2022.02.24 15:53:55 ERROR > bloop.shaded.coursierapi.error.SimpleResolutionError$1: Error downloading org.apache.logging.log4j:log4j-core:2.8.1
2022.02.24 15:53:55 ERROR >   not found: https://artifactory.mydomain.com/artifactory/maven-internalfacing/org/apache/logging/log4j/log4j-core/2.8.1/log4j-core-2.8.1.pom
2022.02.24 15:53:55 ERROR > 	at bloop.shaded.coursierapi.error.SimpleResolutionError.of(SimpleResolutionError.java:11)
2022.02.24 15:53:55 ERROR > 	at bloop.shaded.coursierapi.shaded.coursier.internal.api.ApiHelper$.simpleResError(ApiHelper.scala:326)
2022.02.24 15:53:55 ERROR > 	at bloop.shaded.coursierapi.shaded.coursier.internal.api.ApiHelper$.doFetch(ApiHelper.scala:359)
2022.02.24 15:53:55 ERROR > 	at bloop.shaded.coursierapi.shaded.coursier.internal.api.ApiHelper.doFetch(ApiHelper.scala)
2022.02.24 15:53:55 ERROR > 	at bloop.shaded.coursierapi.Fetch.fetchResult(Fetch.java:244)
2022.02.24 15:53:55 ERROR > 	at bloop.shaded.coursierapi.Fetch.fetch(Fetch.java:239)
2022.02.24 15:53:55 ERROR > 	at bloop.bloopgun.core.DependencyResolution$.resolveWithErrors(DependencyResolution.scala:66)
2022.02.24 15:53:55 ERROR > 	at bloop.bloopgun.core.ServerStatus$.resolveServer(ServerStatus.scala:23)
2022.02.24 15:53:55 ERROR > 	at bloop.bloopgun.BloopgunCli.fireServer(Bloopgun.scala:377)
2022.02.24 15:53:55 ERROR > 	at bloop.bloopgun.BloopgunCli.fireCommand(Bloopgun.scala:283)
2022.02.24 15:53:55 ERROR > 	at bloop.bloopgun.BloopgunCli.run(Bloopgun.scala:228)
2022.02.24 15:53:55 ERROR > 	at bloop.launcher.bsp.BspBridge$$anon$1.run(BspBridge.scala:82)
2022.02.24 15:53:55 ERROR > error: Server could not be started at None:Some(8212)! Giving up...
2022.02.24 15:53:55 ERROR error: The launcher failed to establish a bsp connection, aborting...
2022.02.24 15:53:55 ERROR Failed to connect with build server, no functionality will work.
```

**Expected behavior**

Metals extension should not rely on vulnerable versions of Log4j.

**Installation:**

- Operating system: Mac OSX 11.6.4
- VSCode version: 1.64.2
- VSCode extension version: 1.12.0
- Metals version:  0.11.1

**Search terms**

vscode, bloop, metals, log4j, CVE-2021-44228, CVE-2021-45046, CVE-2021-45105, vulnerability

**Comment by tgodzik** [comment]
Thanks for reporting! Could you try setting the Bloop version in Metals to `1.4.13` ?

**Comment by scottweaver** [comment]
I tried this and it seemed to have no effect.  Metals still attempts to install 1.14.12.

Steps I took:
1. Changed "Metals: Bloop Version` to "1.14.12"
2. Closed VS Code
3. `rm -rf .bloop`
4. `rm -rf .metals`
5. Restarted VS Code
6. Imported the test project.

Still seeing:
```
2022.02.25 09:33:25 ERROR > Resolving ch.epfl.scala:bloop-frontend_2.12:1.4.12...
2022.02.25 09:33:25 ERROR > error: Unexpected error when resolving Bloop server via coursier!
2022.02.25 09:33:25 ERROR > error: Error downloading org.apache.logging.log4j:log4j-core:2.8.1
2022.02.25 09:33:25 ERROR >   not found: https://artifactory.mydomain.com/artifactory/maven-internalfacing/org/apache/logging/log4j/log4j-core/2.8.1/log4j-core-2.8.1.pom
2022.02.25 09:33:25 ERROR > bloop.shaded.coursierapi.error.SimpleResolutionError$1: Error downloading org.apache.logging.log4j:log4j-core:2.8.1
2022.02.25 09:33:25 ERROR >   not found: https://artifactory.mydomain.com/artifactory/maven-internalfacing/org/apache/logging/log4j/log4j-core/2.8.1/log4j-core-2.8.1.pom
2022.02.25 09:33:25 ERROR > 	at bloop.shaded.coursierapi.error.SimpleResolutionError.of(SimpleResolutionError.java:11)
```

**Comment by tgodzik** [comment]
> Changed "Metals: Bloop Version` to "1.14.12"

Could you try setting it to `1.4.13` instead of `1.4.12` ?

**Comment by scottweaver** [comment]
Sorry, that's what I meant :D

**Comment by tgodzik** [comment]
That might be Metals dependency actually. Could you try using the latest prerelease version to see if that helps? It can be changed on the extension page in VS Code.

**Comment by scottweaver** [comment]
I tried switching to pre-release yesterday and that didn't address the issue.

**Comment by tgodzik** [comment]
Ok, let's try switching Bloop in Metals to 1.4.13, this should cause 1.4.12 version to be downloaed.

https://github.com/scalameta/metals/pull/3664

Was the message the same when you tried it?

**Comment by scottweaver** [comment]
Ok looks like this works:

1. Set `Metals: Bloop Version` to `1.4.13` under `Preferences > Settings filter on "Metals"`.
2. Changed Metals to the Pre Release version via the extensions page in VS Code.
3. Reload Window.
4. Import Build.
5. Bloop now installs without issue.

Extra steps I took was to `rm -rf .bloop` and `rm -rf .metals` (done between steps 1 and 2).  This may not be necessary.

Thanks for the help @tgodzik!
